binding/nodejs/index.js

```diff
@@ -183,29 +144,22 @@ class Porcupine {
       );
     }
 
-    const frameBuffer = int16Array(this.frameLength);
-    for (let i = 0; i < this.frameLength; i++) {
-      frameBuffer[i] = frame[i];
-    }
-
-    const status = this.libPpn.pv_porcupine_process(
-      this.ppnPtr,
-      frameBuffer,
-      this.keywordIndexPtr
-    );
+    const frameBuffer = new Int16Array(frame);
 
+    const packed = pv_porcupine.process(this.handle, frameBuffer);
+    const status = packed % 10;
+    const keywordIndex = (status == PV_STATUS_T.SUCCESS) ? (packed / 10) : -1;
     if (status !== PV_STATUS_T.SUCCESS) {
       pvStatusToException(status, "Porcupine failed to process the frame");
```

Comment:
same as before (move the status/exception check above the keywordIndex assignment) , unless I'm missing something